src/QsCompiler/TextProcessor/QsExpressionParsing.fs

```diff
@@ -462,13 +462,17 @@ let private valueTuple item = // allows something like (a,(),b)
 
 /// Parses a Q# value array as QsExpression.
 /// Uses commaSep1 to generate suitable errors for invalid or missing expressions within the array.
-let private valueArray = // this disallows []
-    let content =
-        let items =
-            commaSep1 expr ErrorCode.InvalidExpression ErrorCode.MissingExpression unknownExpr eof
-            |>> ValueArray
+let private valueArray =
+    let sized = expr .>>. (comma >>. size.parse >>. equal >>. expectedExpr rArray) |>> SizedArray
+
+    // This disallows [].
```

Comment:
The [QEP ](https://github.com/microsoft/qsharp-language/blob/main/Approved/2-enhanced-array-literals.md) mentions that the empty array literal will be supported, did something change that made `[]` disallowed?

Reply:
I'm working on support for `[]` in a separate branch right now. It's a larger change that is independent from the `size = n` feature, so it will be in a separate PR.